test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart

```diff
@@ -23,137 +22,114 @@ part 'constant_required_accessor_integration_test.over_react.g.dart';
           )()
-        );
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+          );
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
-
     });
 
     group('throwing when a prop is required and set to null', () {
       test('on mount', () {
-        render((ComponentTest()
+        expect(() {
+          mount((ComponentTest()
             ..required = null
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
           )());
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
 
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
-            ..required = true
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
+              attachedToDocument: true);
+        }, logsNoPropTypeWarnings);
+
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = null
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
-          )(),
-          attachedToDocument: true,
-        );
-
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
-
-        jacket.rerender((ComponentTest()
-              ..required = null
-              ..nullable = true
-              ..requiredAndLengthLimited = [1,2]
-            )());
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+          )());
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
     });
 
     group('throwing when a prop is nullable and not set', () {
       test('on mount', () {
-        render((ComponentTest()
-          ..required = true
-          ..requiredAndLengthLimited = [1,2]
-        )());
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.nullable')]);
+        expect(() {
+          mount((ComponentTest()
+            ..required = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsPropError('ComponentTestProps.nullable'));
       });
 
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```

Comment:
#nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```